src/store/reducer/counterSlice.ts

```diff
@@ -0,0 +1,42 @@
+import { createSlice, PayloadAction } from "@reduxjs/toolkit";
+
+import type { AppState, AppThunk } from "../index";
+
+export interface CounterState {
+  value: number;
+  status: "idle" | "loading" | "failed";
+}
+
+const initialState: CounterState = {
+  value: 0,
+  status: "idle",
+};
+
+export const counterSlice = createSlice({
+  name: "counter",
+  initialState,
+  reducers: {
+    increment: (state) => {
+      state.value += 1;
+    },
+    decrement: (state) => {
+      state.value -= 1;
+    },
+    incrementByAmount: (state, action: PayloadAction<number>) => {
+      state.value += action.payload;
+    },
+  },
+});
+
+export const { increment, decrement, incrementByAmount } = counterSlice.actions;
+export const selectCount = (state: AppState) => state.counter.value;
+export const incrementIfOdd =
+  (amount: number): AppThunk =>
+  (dispatch, getState) => {
+    const currentValue = selectCount(getState());
+    if (currentValue % 2 === 1) {
+      dispatch(incrementByAmount(amount));
+    }
+  };
+
+export default counterSlice.reducer;
```

Comment:
What is this counter slice for? At least have it on the index page or delete it.

Reply:
It is a placeholder. if it be deleted, next will report an error saying that the reducer is missing from the store